src/utils.ts

```diff
@@ -0,0 +1,56 @@
+import * as core from '@actions/core'
+import {AxiosError, AxiosInstance, AxiosRequestConfig, AxiosStatic} from 'axios'
+import fs from 'fs'
+
+export function debugAxiosError(error: AxiosError): void {
+  const debug: string =
+    `Request on ${error.config.url} failed with ` +
+    `code: ${error.code} ` +
+    `status: ${error.response?.status} ` +
+    `data: ${error.response?.data}`
+  core.info(debug)
+}
+
+export function retryArtifacts(error: AxiosError): boolean {
+  return (
+    error.code !== 'ECONNABORTED' &&
+    (!error.response ||
+      (error.response.status >= 500 && error.response.status <= 599))
+  )
+}
+
+export function exponentialDelay(retryNumber = 0): number {
+  const delay = Math.pow(2, retryNumber) * 100
+  const randomSum = delay * 0.2 * Math.random()
+  return delay + randomSum
+}
+
+export function artifactsRetry(
```

Comment:
This is a similar approach into how the axios retry actually works, now we have a bit more control over it